### ISSUE TITLE
The no_progress_bar option doesn't work

### ISSUE THREAD

**Issue by frankang** [facebookresearch/fairseq-py#17]
Hi, on two of my computer (Ubuntu 16.04 and14.04), the no-progress-bar option does not work as expected, the program still prints a new line for each iteration. This option only works on another Debian system computer, where the script prints the status according to the log-interval option.

And while I keep the progress bar enabled (no-progress-bar=False), and replace _progress_bar_ in the train.py file with the native _tqdm_ object (with options, as demonstrated below), a very fast print of the status can still be observed.
```
 with progress_bar(itr, desc, leave=False) as t: #original
 with tqdm(itr, desc, leave=False,miniters=1000,mininterval=10.0) as t: #replaced
```

I suspect that could be an issue of the tqdm tool. My current solution is to add an option _refresh=False_ to the "t.set_postfix(collections.OrderedDict)" part in the _train.py_ file. 
t.set_postfix(collections.OrderedDict(),refresh=False)
After that, there is no output of the progress-bar at every iteration, which is the expected behavior when no_progress_bar option is True.

Could anyone suggest the possible reason for this issue? Thanks.

**Comment by myleott** [comment]
I'm not sure I've seen this before. What version of tqdm and Python do you have installed? Are you using a special terminal or job scheduling system (e.g., Slurm)?

**Comment by frankang** [comment]
I followed the installation guide as described in this project's readme file. The tqdm version is "4.19.1.post1" and "4.17.1" on the two problem-free computers. And on the computer without problem the tqdm has a version of 4.15.0. Python version is 3.6.1

I'm using MobaXterm on Windows (MobaX Xorg version), with SSH connection to those computers. I'm not using any job scheduling system.
I ran the following code on all my systems and didn't get any error. A normal working progress bar is correctly displayed.

`python #enter python runtime
import time
from tqdm import tqdm
for i in tqdm(range(10000),leave=True,miniters=2):
    time.sleep(1)
`
